packages/react-swipeable-views/src/SwipeableViews.js

```diff
@@ -301,18 +301,19 @@ class SwipeableViews extends React.Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { index } = nextProps;
+  componentDidUpdate(prevProps) {
+    const { index } = this.props;
 
-    if (typeof index === 'number' && index !== this.props.index) {
+    if (typeof index === 'number' && index !== prevProps.index) {
       if (process.env.NODE_ENV !== 'production') {
-        checkIndexBounds(nextProps);
+        checkIndexBounds(this.props);
       }
 
       this.setIndexCurrent(index);
+      // eslint-disable-next-line react/no-did-update-set-state
       this.setState({
-        // If true, we are going to change the children. We shoudn't animate it.
-        displaySameSlide: getDisplaySameSlide(this.props, nextProps),
+        // If true, we are going to change the children. We shouldn't animate it.
+        displaySameSlide: getDisplaySameSlide(prevProps, this.props),
```

Comment:
Hm, on further testing, it *seems* to work fine, but on a project I'm using it with it fails the first transition. Might be something with newer versions of React, but the "displaySameSlide" calculation happens too late on my project using 16.12 and not the version in this repo. Going to see if I can somehow move the calculation out or conditionally set it to false initially.